session-8/model/defining/task2.go

```diff
@@ -0,0 +1,31 @@
+package defining
+
+import "fmt"
+
+type Speaker interface {
+	Speak() string
+}
+type Dog struct {
+	Name string
+}
+type Person struct {
+	Name string
+}
+
+func (d Dog) Speak() string {
+	return fmt.Sprintf("Woof!")
+}
+func (p Person) Speak() string {
+	return fmt.Sprintf("Hello!")
+}
+
+func Task2() {
+	fmt.Println("Task 2  ****************")
+	var dog Speaker
+	dog = Dog{Name: "Dogy"}
+
+	var person Speaker
+	person = Person{Name: "John"}
```

Comment:
Eger her hansisa variable-da dinamik olaraq data assign edib, sonradan deyishmirikse ve her hansisa shertlerde ferqli implemenntasiya etmirikse, o zaman variable-lari ayrica elan edib sonradan assign etmesek de olar.
```suggestion
	dog := Dog{Name: "Dogy"}
	person := Person{Name: "John"}
```